students/tfbanks/Lesson06/Mailroom_pt4.py

```diff
@@ -0,0 +1,90 @@
+# Mailroom_pt4.py Exercise by tfbanks
+
+# !/usr/bin/env python3
+
+# pt4 update - I made some changes to clean up some code as well as broke out thank_you_letter so more easily test that function
+
+# Donor Database containing names and donations
+donor_names = {
+    'Tim Cooker': [1500.50, 25050.50, 15680.75],
+    'Elon Musket': [5250.25, 26500],
+    'Frank Petersmankempt': [550.60],
+    'Megan Morgan': [650.40, 1600, 20625.40],
+    'Marlene Wheeler': [820, 1222.80]}
+
+
+def selection(option):
+    switch_fun_dict = {1: thank_you_donor, 2: report, 3: mass_mail}
+    return switch_fun_dict[option]
+
+
+def thank_you_donor():  # Code for selecting Donor and writing a Thank You Note
+    ty_donor = input('Enter the Full Name of a donor (for list of previous donors type list): ').title()
+    if ty_donor == 'List':  # If List, Prints Donor List and restarts the function
+        for donor in donor_names.keys():
+            print(donor)
+        ty_donor = input('Enter the Full Name of a donor (for list of previous donors type list): ').title()
+
+    while True:
+        try:
+            amount = float(input('Please enter the donation amount: '))
+        except ValueError:
+            print('Please enter a valid donation amount')
+        else:
+            break
+    if ty_donor in donor_names:  # Actions if donor is already in donor_names
+        donor_names[ty_donor].append(amount)  # Appends donation amount to donators donation values
+    else:
+        donor_names[ty_donor] = [amount]
+    return thank_you_letter(ty_donor, amount)
+
+
+def thank_you_letter(ty_donor, amount):
+    letter = ('''\n\nDear {},\n
+Thank you for your generous donation of ${:,.2f}, your generosity is greatly appreciated.
+These funds will help continue our efforts to teach Python to the next generation.\n
+Sincerely,\n\n
+Joesef Edword Bringingham\n\n''')
+    print(letter.format(ty_donor, amount))
```

Comment:
to unit test have a function to generate and return the string then separate function to print.